pkg/address/validate_address.go

```diff
@@ -0,0 +1,59 @@
+package address
+
+import (
+	"fmt"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/zeta-chain/zetacore/pkg/chains"
+)
+
+// ValidateAddressForChain validates the address for the chain
+// NOTE: since these checks are currently not used, we don't provide additional chains for simplicity
+// https://github.com/zeta-chain/node/issues/2234
+// https://github.com/zeta-chain/node/issues/2385
+// NOTE: We should eventually not using these hard-coded checks at all for same reasons as above
+
+// TODO : Use this function to validate Sender and Receiver address for CCTX
+// https://github.com/zeta-chain/node/issues/2697
+func ValidateAddressForChain(address string, chainID int64, additionalChains []chains.Chain) error {
+	chain, found := chains.GetChainFromChainID(chainID, additionalChains)
+	if !found {
+		return fmt.Errorf("chain id %d not supported", chainID)
+	}
+	switch chain.Network {
+	case chains.Network_eth:
+		return ValidateEthereumAddress(address)
+	case chains.Network_zeta:
+		return nil
+	case chains.Network_btc:
+		{
+			addr, err := chains.DecodeBtcAddress(address, chainID)
+			if err != nil {
+				return fmt.Errorf("invalid address %s , chain %d: %s", address, chainID, err)
+			}
+			if !chains.IsBtcAddressSupported(addr) {
+				return fmt.Errorf("unsupported address %s", address)
+			}
+			return nil
+		}
+	case chains.Network_polygon:
+		return ValidateEthereumAddress(address)
+	case chains.Network_bsc:
+		return ValidateEthereumAddress(address)
+	case chains.Network_optimism:
+		return nil
```

Comment:
I think we need to validate by VM type, not by chain.network, all EVM's have the same addr

Reply:
Yes, I agree, to add all new check here we would need adding a new VM for Bitcoin (which can make sense as Bitcoin has some scripting functionality) but this would be higher scope than the PR.
I suggest to use VM for check with an exception for Bitcoin networks

Reply:
I agree with the fact the function should check based on the VM, which is in fact what this function does internally . 
All EVM based chains can use the same `ValidateEthereumAddress` function. 

Adding individual cases for each chain lets us provide a more granular check for each chain , and choose to ignore checks for some chains if we want to. 
IMO for checks like these it would be good be more cautious , and keep it at chain level